db/src/cachedb.rs

```diff
@@ -35,13 +112,13 @@ impl<T> KeyValueDB for CacheDB<T>
 where
     T: KeyValueDB,
 {
-    type Batch = T::Batch;
+    type Batch = CacheDBBatch<T::Batch>;
 
     fn read(&self, col: Col, key: &[u8]) -> Result<Option<Vec<u8>>> {
-        let cache_guard = self.cache.read();
+        let mut cache_guard = self.cache.lock();
```

Comment:
each read will hold the lock of cache and block other read operations, even it's reading the none cachable cols, we may need to change `CacheTable`
```
type CacheTable = FnvHashMap<Col, Mutex<LruCache<Vec<u8>, Vec<u8>>>>;
```